src/components/Filters/Filters.tsx

```diff
@@ -0,0 +1,179 @@
+import React from 'react';
+import isEqual from 'lodash/isEqual';
+
+import {block} from '../utils/cn';
+import {FieldControl} from './components/FilterControl';
+import {FilterSelect} from './components/FilterSelect';
+import {ClearFiltersButton} from './components/ClearFiltersButton/ClearFiltersButton';
+
+import i18n from './i18n';
+
+import type {FilterSchema} from './schemas/types';
+
+import './Filters.scss';
+
+const b = block('filters');
+
+export type FilterValue<TFilterValue extends any = string[]> = TFilterValue;
+export type FiltersValue<TFiltersValue extends any = Record<string, FilterValue>> = TFiltersValue;
+
+/**
+ * Filters schema to retreive editors/initial values/other schema options.
+ */
+export type FiltersSchema = Record<string, FilterSchema<any, any>>;
+
+interface FiltersComponentProps {
+    className?: string;
+
+    filtersSchema: FiltersSchema;
+
+    showEmptyText?: boolean;
+
+    initialOpenFilterKey?: string;
+
+    /**
+     * Contains serialized filters values with ID as a key.
+     * Filters component uses this object to render selected filters.
+     *
+     * @example {location: ['Moscow'], author: ['Alex']}
+     */
+    initialValues?: FiltersValue;
+    /**
+     * Contains serialized filters values with ID as a key.
+     * Filters component uses this object to render selected filters.
+     *
+     * @example {location: ['Moscow', 'London'], author: ['Max', 'Travis']}
+     */
+    filtersValues: FiltersValue;
+
+    handleChangeFilterValue: (id: string, value: FilterValue) => void;
+
+    handleRemoveFilterValue: (id: string, value: FilterValue) => void;
+
+    /**
+     *
+     * @param values Currently applied filters
+     */
+    handleClearAllFilters: (values: React.ComponentProps<typeof Filters>['filtersValues']) => void;
+}
+
+/**
+ *
+ * @description Current version of Filters component is unstable. It's currently being tested.
+ */
+export function Filters(props: FiltersComponentProps) {
+    const {
+        className,
+        filtersSchema,
+        showEmptyText = false,
+        initialOpenFilterKey,
+        initialValues,
+        filtersValues = {},
```

Comment:
U have `filtersValues` as a dependency in useMemo. It is not a good idea to have such default value